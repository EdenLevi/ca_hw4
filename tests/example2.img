# Test 2

L4 #load latency
S1 #store latency
O1 #overhead for switch
N3  #there is three threads that run 

T0 
I@0x00000000
LOAD $1, $0, 0x32A0
LOAD $2, $0, 0x32A4
SUB $3, $2, $1
HALT $0

T1
I@0x00000000
LOAD $5, $0, 0x32A0 # 0x0
ADD $5, $5, $5     # 0x4
ADD $2, $5, $5    # 0x8
HALT $0

T2
I@0x00000000
LOAD $5, $0, 0x32A0  # 0x0
ADDI $7, $5, 5    # 0x4
ADD $2, $5, $5    # 0x8
HALT $0

# Data for this program. Any other location that is not explicitly set is implicitly set to zero.
# All the threads share the same memory space
D@0x000032A0
0x4
0x10

