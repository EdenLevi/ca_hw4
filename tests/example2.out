
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0x4	R2 = 0x10	R3 = 0xC	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x10	R3 = 0x0	R4 = 0x0	R5 = 0x8	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x8	R3 = 0x0	R4 = 0x0	R5 = 0x4	R6 = 0x0	R7 = 0x9
Blocked MT CPI for this program 1.500000

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x4	R2 = 0x10	R3 = 0xC	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x10	R3 = 0x0	R4 = 0x0	R5 = 0x8	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x8	R3 = 0x0	R4 = 0x0	R5 = 0x4	R6 = 0x0	R7 = 0x9
Finegrained Multithreading CPI for this program 1.166667

